### PR TITLE
1343 - Added 'listen_addresses' endpoint.

### LINKED ISSUE #1343: /network/listen_addresses endpoint
/network/listen_addresses

Get a list of libp2p multiaddr the node is listening on.

Lighthouse: https://lighthouse-book.sigmaprime.io/http_network.html#networklisten_addresses

### PR DESCRIPTION
**Pull request by mark-terry**
## PR Description
Added the 'listen_addresses' endpoint.

## Fixed Issue(s)
Fixes #1343

### CODE REVIEW

File: data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/handlers/network/GetListenAddresses.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  @OpenApi(
+      path = ROUTE,
+      method = HttpMethod.GET,
+      summary = "Returns the list of MultiAddr that the client's libp2p service is listening on.",
+      tags = {TAG_NETWORK},
+      description =
+          "Returns the list of MultiAddr that the client's libp2p service is listening on.",
+      responses = {
+        @OpenApiResponse(status = RES_OK, content = @OpenApiContent(from = String.class)),
```
Comment (rolfyone) [inline data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/handlers/network/GetListenAddresses.java]: should really be 
```
@OpenApiResponse(status = RES_OK, content = @OpenApiContent(from = String.class, isArray = true))
```

File: data/provider/src/main/java/tech/pegasys/artemis/api/NetworkDataProvider.java [hunk 2]
```diff
@@ -79,4 +80,8 @@ public int getListenPort() {
   P2PNetwork<?> getP2pNetwork() {
     return p2pNetwork;
   }
+
+  public List<String> getListeningAddresses() {
+    return List.of(LibP2PNetwork.getAdvertisedAddrString(p2pNetwork.getConfig()));
+  }
```
Comment (rolfyone) [inline data/provider/src/main/java/tech/pegasys/artemis/api/NetworkDataProvider.java:86]: nit: can we have a unit test for this?

Comment (mark-terry) [reply data/provider/src/main/java/tech/pegasys/artemis/api/NetworkDataProvider.java:86]: Added.

**Review by rolfyone — APPROVED** [review]
LGTM

**Pull request merged**